lua/lean/html.lua

```diff
@@ -357,7 +357,8 @@ function BufDiv:new(buf, div, keymaps)
   util.set_augroup("DivPosition", string.format([[
     autocmd CursorMoved <buffer=%d> lua require'lean.html'._by_id[%d]:buf_update_cursor()
     autocmd BufEnter <buffer=%d> lua require'lean.html'._by_id[%d]:buf_update_cursor()
-  ]], buf, id, buf, id), buf)
+    autocmd BufDelete <buffer=%d> lua require'lean.html'._by_id[%d] = nil
```

Comment:
From what I can tell this has no effect, that is when I open a tooltip and close it repeatedly and do

```
:lua for id, _ in pairs(require'lean.html'._by_id) do print(vim.inspect(id)) end
```

that list only ever grows, which makes me suspect refs aren't getting released as they should. Not sure what's going on, perhaps we need a different autocmd.

Reply:
Yes I've confirmed that refs are no longer getting released at all if the `BufDiv` corresponding to a tooltip survives. In general this seems like a more fragile way of doing things, could we instead make a local `BufDiv` in `commands.lua` that is `nil`'ed when the popup is closed?

Reply:
Good catch. Okay, I've switched back to the weak table and put the `bufdiv` variable in `commands.lua`.  The downside is that you can't have tooltips open in multiple windows at the same time.

> nil'ed when the popup is closed?

Unfortunately, I couldn't find any reliable way to either detect when the floating window is closed or when the buffer is deleted.  Autocmds like `BufDelete`, `BufUnload`, `WinClose` don't seem to trigger for this popup???  Weirdly enough, `BufUnload` triggers for the tooltips in the infoview (but not `BufDelete`).

If we could reliably detect when a buffer is deleted, then I think this would be the cleanest solution here.  You wouldn't need to keep a reference to the `BufDiv` object (which is awkward for popups), and it doesn't introduce any leaks either (it only leaks the BufDiv if the buffer is leaked).

Reply:
> If we could reliably detect when a buffer is deleted, then I think this would be the cleanest solution here.

I agree, but it looks like we're running into upstream issues, though I haven't found any existing reports on this. So I guess we'll have to deal with it for now.